Plugin/Sales/Model/Spi/CreditmemoResource.php

```diff
@@ -193,12 +165,14 @@ public function aroundLoad(
         /** @var \Magento\Framework\Model\ResourceModel\Db\AbstractDb $resultEntity */
         $resultEntity = $proceed($entity, $value, $field);
 
-        // Load AvaTax extension attributes
+        // Load AvaTax record into extension attributes
         if ($this->avaTaxConfig->isModuleEnabled($entity->getStoreId())) {
 
-            // Get the AvaTax Attributes from the AbstractModel
-            $avataxIsUnbalanced = $entity->getData('avatax_is_unbalanced');
-            $baseAvataxTaxAmount = $entity->getData('base_avatax_tax_amount');
+            // Get the AvaTax record
+            /** @var CreditMemo $avataxRecord */
+            $avataxRecord = $this->avataxCreditMemo->loadByParentId($entity->getId());
```

Comment:
I think you can call load() on the model with an additional parameter to specify the field you want to load by, and thus eliminate the loadByParentId() methods on both your model and resourceModel classes.

`$this->avataxCreditMemo->load($entity->getId(), AvaTaxCreditMemoResource::PARENT_ID_FIELD_NAME);`

Reference: \Magento\Framework\Model\ResourceModel\Db\AbstractDb::load